### ISSUE TITLE
[opencv[cuda]:x64-windows] build failure

### ISSUE THREAD

**Issue by h101059** [microsoft/vcpkg#12539]
**Host Environment**
- OS: Windows 10
- Compiler: Visual Studio Professional 2019 16.6.5

**To Reproduce**
`./vcpkg install opencv[cuda]:x64-windows --recurse`

**Failure logs** 
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: ninja -v
    Working Directory: C:/Repos/vcpkg/buildtrees/opencv4/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\Repos\vcpkg\buildtrees\opencv4\config-x64-windows-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_cmake.cmake:297 (vcpkg_execute_required_process)
  ports/opencv4/portfile.cmake:262 (vcpkg_configure_cmake)
  scripts/ports.cmake:90 (include)

Error: Building package opencv4:x64-windows failed with: BUILD_FAILED

**Additional context**
A couple of months ago I used the command `./vcpkg install opencv:x64-windows-static' to install opencv library, the opencv version 4.1.1 was installed successfully. Today I tried to install the 'cuda' feature on top of it and it failed on both command `./vcpkg install opencv[cuda]:x64-windows-static --recurse` and `./vcpkg install opencv[cuda]:x64-windows --recurse` with the same error.

**Comment by StarGate-One** [comment]
Please attach/upload `C:\Repos\vcpkg\buildtrees\opencv4\config-x64-windows-out.log` to allow for additional review and solution to this issue.

**Comment by h101059** [comment]
[config-x64-windows-static-out.log](https://github.com/microsoft/vcpkg/files/4963177/config-x64-windows-static-out.log)

**Comment by h101059** [comment]
[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/4963180/config-x64-windows-out.log)

**Comment by h101059** [comment]
I have CUDA toolkit version 11.0 installed on my machine.

**Comment by StarGate-One** [comment]
The failure in both the static and dynamic builds are the same:
`CMake Generate step failed.  Build files cannot be regenerated correctly.`
`ninja: build stopped: subcommand failed.`

I am not sure what causes CMake to fail on this. Maybe someone with more experience can assist.

I know sometimes, but no guarantee as I do not use opendv, the port can be built if you comment out the PREFER_NINJA in the `<vcpkg-root>\ports\opencv4\portfile.cmake` like this:

`vcpkg_configure_cmake(`
`    # PREFER_NINJA`
`    SOURCE_PATH ${SOURCE_PATH}`

**Comment by JackBoosY** [comment]
Duplicate with #7913, please track that issue.

Thanks.

**Comment by h101059** [comment]
After commenting out the PREFER_NINJA, the build fails the same way without  commenting out the PREFER_NINJA. Thanks for the suggestion.

**Comment by h101059** [comment]
I found that the build errors are caused by missing the "nppicom.lib" library file inside CUDA toolkit v11.0 version. Fortunately, the  CUDA toolkit version 11.0 installation contains version v10.2, v10.0, and v9.0 as well. So in order to allow vcpkg.exe locates the latest version of CUDA toolkit SDK as v10.2, I changed the CUDA folder directory "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\11.0" into "C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\1.10" and applied command "./vcpkg install opencv[cuda]:x64-windows-static --recurse" to rebuilt it. This time the build succeeds.

It looks like there is a bug inside CUDA toolkit version 11.0.
